model/utils.go

```diff
@@ -573,3 +573,26 @@ func IsDomainName(s string) bool {
 
 	return ok
 }
+
+func RemoveDuplicateStrings(in []string) []string {
```

Comment:
I'm not entirely convinced we want to export this in the `model` library, since this is something consumer by other Mattermost libraries and such. Can we just inline this where it's needed for now?

If we don't want to make it a general purpose utility function, we should minimally allocate the `seen` map to have a capacity matching the `in` slice.

Reply:
this will be part of this https://github.com/mattermost/mattermost-server/blob/scrolling-overhaul/model/utils.go#L577 I'm just reusing here per @hmhealey suggestion

Reply:
Ah, interesting. Can you change it to pre-allocate the `seen` map, under the assumption that the size of the resulting output is often very close to the original:
```
	seen := make(map[string]bool, len(in))
```

(Knowing when to do this or not is one of the reasons I'd generally avoid writing this as a global helper method, since there may very well be cases where lots of duplicates are expected, but this method will assume the otherwise.)

Reply:
Also, @hmhealey, thoughts re: above conversation?

Reply:
did the change @hmhealey and @lieut-data PTAL